models/user.py

```diff
@@ -1,38 +1,90 @@
-from models.BaseModel import BaseModel
+from utility.helpers import query
+from utility.decor import path_set
+from flask import g
 
 
-class User(BaseModel):
 
+####
+class User():
     def __init__(self):
-        self.email_userid = {}
-        super().__init__()
+        pass
 
-    def add(self, entry):
-        userid = super().add(entry)
-        email = entry['email']
-        self.email_userid[email] = userid
+    @path_set(path='SQL/user')
+    def add(self, client_name, email, password, path):
 
-    def validate(self, email, password):
+        query(path, 'register',
+              vals=(client_name, email, password))
 
-        if self.old_user(email):
-            userid = self.email_userid[email]
-            entry = self.find_val(userid)
-            return entry['password'] == password
+    @path_set(path='SQL/user')
+    def validate(self, email, password, path):
+
+        query(path, 'validate', vals=(email,))
+        cur = g.cur 
+        row = cur.fetchone()
+
+        if row is None or row==[]:
+            return False
         else:
-            False
+            inquired_password = row['passkey']
+            return password == inquired_password
+
+    @path_set(path='SQL/user')
+    def old_user(self, email, path):
+
+        query(path, 'old_user' , vals=(email,))
+        cur = g.cur 
+        appear_num = cur.fetchall()
+        return appear_num != 0
+
+    @path_set(path='SQL/user')
+    def delete(self, userid, path):
+
+        query(path,  'delete', vals=(userid,))
+
+    @path_set(path='SQL/user')
+    def find_userid_by_email(self, email, path):
```

Comment:
This should be `find_by_email` that returns a full dictionary. Then the caller can extract whatever data it wants from the dic.